src/themes/material/components/edit-chip-list/edit-chip-list.component.ts

```diff
@@ -0,0 +1,198 @@
+import {
+	Component,
+	Input,
+	OnInit,
+	ChangeDetectorRef,
+	ViewChildren,
+	QueryList
+} from '@angular/core';
+
+import {AppError} from 'ng2-qgrid/core/infrastructure';
+import {Command, CommandManager} from 'ng2-qgrid/core/command';
+import {Shortcut, ShortcutManager} from 'ng2-qgrid/core/shortcut';
+import {EditChipComponent} from './edit-chip/edit-chip.component';
+import {ChipState} from './edit-chip/edit-chip.component';
+
+interface ISelected {
+	index?: number;
+	value?: string;
+	action: Actions;
+}
+
+interface IChipItem {
+	value?: string | any;
+	state?: string;
+	qGridFocusEnabled?: boolean;
+}
+
+enum Actions {
+	add,
+	delete,
+	focusNext,
+	focusPrevious,
+	inactive,
+}
+
+@Component({
+	selector: 'q-grid-chip-list',
+	templateUrl: './edit-chip-list.tpl.html',
+	styleUrls: ['./edit-chip-list.scss']
+})
+export class EditChipListComponent implements OnInit {
+	@Input('values') chipValues: any[];
+
+	@ViewChildren(EditChipComponent)
+	private chipList: QueryList<EditChipComponent>;
+
+	private items: IChipItem[];
+
+	private shortcut = new Shortcut(new ShortcutManager());
+
+	private selectedItem: ISelected = {
+		action: Actions.inactive
+	};
+
+	private incrementSelectedIndex() {
+		if (this.selectedItem.index === this.chipValues.length) {
```

Comment:
bad names and too complicated, selectedItem.index is a confusion part, too much state in one variable

Reply:
Have refactored this in the commit: [Merge from Dev: conflicts resolving](https://github.com/qgrid/ng2/pull/39/commits/da05f15a228a611a5059e3049c9f18402446fa8f).